multicluster/test/e2e/service_test.go

```diff
@@ -156,27 +175,178 @@ func (data *MCTestData) testANPToServices(t *testing.T) {
+
+	connectivity = data.probeFromPodInCluster(eastCluster, multiClusterTestNamespace, eastRegularClientName, "client", eastIP, mcWestClusterTestService, 80, corev1.ProtocolTCP)
+	assert.Equal(t, antreae2e.Dropped, connectivity, "Failure -- wrong result from probing exported Service from regular clientPod after applying toServices AntreaNetworkPolicy")
+}
+
+func (data *MCTestData) testStretchedNetworkPolicy(t *testing.T) {
+	westExpSvc, err := data.getService(eastCluster, multiClusterTestNamespace, mcWestClusterTestService)
+	if err != nil {
+		t.Fatalf("Error when getting the imported Service %s: %v", mcWestClusterTestService, err)
+	}
+	westExpSvcIP := westExpSvc.Spec.ClusterIP
+	eastGwClientName := getClusterGatewayClientPodName(eastCluster)
+	eastRegularClientName := getClusterRegularClientPodName(eastCluster)
+
+	// Verify that Stretched NetworkPolicy works fine with podSelect or podSelect+nsSelector.
+	acnpBuilder1 := &e2euttils.ClusterNetworkPolicySpecBuilder{}
+	acnpBuilder1 = acnpBuilder1.SetName("drop-client-pod-sel").
+		SetPriority(1.0).
+		SetAppliedToGroup([]e2euttils.ACNPAppliedToSpec{{PodSelector: map[string]string{"app": "nginx"}}}).
+		AddStretchedIngressRule(map[string]string{"antrea-e2e": eastGwClientName}, nil, "", nil, crdv1alpha1.RuleActionDrop).
+		AddStretchedIngressRule(map[string]string{"antrea-e2e": eastRegularClientName}, map[string]string{"kubernetes.io/metadata.name": multiClusterTestNamespace}, "", nil, crdv1alpha1.RuleActionDrop)
+	if _, err := data.createOrUpdateACNP(westCluster, acnpBuilder1.Get()); err != nil {
+		t.Fatalf("Error creating ACNP %s: %v", acnpBuilder1.Name, err)
 	}
 
-	connectivity = data.probeFromPodInCluster(eastCluster, multiClusterTestNamespace, eastRegularClientName, "client", eastIP, fmt.Sprintf("antrea-mc-%s", westClusterTestService), 80, corev1.ProtocolTCP)
-	if connectivity == antreae2e.Error {
-		t.Errorf("Failure -- could not complete probeFromPodInCluster: %v", err)
-	} else if connectivity != antreae2e.Dropped {
-		t.Errorf("Failure -- wrong result from probing exported Service from regular clientPod after applying toServices AntreaNetworkPolicy. Expected: %v, Actual: %v", antreae2e.Dropped, connectivity)
+	connectivity := data.probeFromPodInCluster(eastCluster, multiClusterTestNamespace, eastGwClientName, "client", westExpSvcIP, mcWestClusterTestService, 80, corev1.ProtocolTCP)
+	assert.Equal(t, antreae2e.Dropped, connectivity, "Failure -- wrong result from probing exported Service from gateway clientPod after applying Stretched NetworkPolicy")
+
+	connectivity = data.probeFromPodInCluster(eastCluster, multiClusterTestNamespace, eastRegularClientName, "client", westExpSvcIP, mcWestClusterTestService, 80, corev1.ProtocolTCP)
+	assert.Equal(t, antreae2e.Dropped, connectivity, "Failure -- wrong result from probing exported Service from regular clientPod after applying Stretched NetworkPolicy")
+	data.deleteACNP(westCluster, acnpBuilder1.Name)
```

Comment:
You may define the error message as below, and call it in related lines.
```
func getErrorMessge (client string) string {
  return fmt.Printf("Failure -- wrong result from probing exported Service from %s clientPod after applying Stretched NetworkPolicy", client)
}
```

Reply:
Done.